src/Storage/TSDatabaseView.m

```diff
@@ -265,11 +266,10 @@ + (void)asyncRegisterSecondaryDevicesDatabaseView
 + (NSDate *)localTimeReceiveDateForInteraction:(TSInteraction *)interaction {
     NSDate *interactionDate = interaction.date;
 
-    if ([interaction isKindOfClass:[TSIncomingMessage class]]) {
-        TSIncomingMessage *message = (TSIncomingMessage *)interaction;
-
-        if (message.receivedAt) {
-            interactionDate = message.receivedAt;
+    if ([interaction isKindOfClass:[TSMessage class]]) {
```

Comment:
> For me anyway, the new implementation is clearer:

There can be no doubt. I was as usual erring on the side of the more minimal change since I still feed new to this codebase, but doing it "the right way" was a much better idea.